tests/test_mq_service.py

```diff
@@ -1080,13 +1080,45 @@ def test_no_identity_and_no_rhsm_reporter(mocker, event_datetime_mock, flask_app
     mock_event_producer = mocker.Mock()
 
     host = minimal_host(account=SYSTEM_IDENTITY["account_number"], insights_id=expected_insights_id)
-    message = wrap_message(host.data())
+
+    platform_metadata = get_platform_metadata()
+    platform_metadata.pop("b64_identity")
+
+    message = wrap_message(host.data(), "add_host", platform_metadata)
 
     with pytest.raises(ValueError):
         handle_message(json.dumps(message), mock_event_producer)
 
 
 # Adding a host requires identity or rhsm-conduit reporter, which does not have identity
+def test_rhsm_reporter_and_no_platform_metadata(mocker, event_datetime_mock, flask_app):
+    expected_insights_id = generate_uuid()
+    host_id = generate_uuid()
+
+    mocker.patch(
+        "app.queue.queue.add_host",
+        return_value=(
+            {"id": host_id, "insights_id": expected_insights_id},
+            host_id,
+            expected_insights_id,
+            AddHostResult.created,
+        ),
+    )
+    mock_event_producer = mocker.Mock()
+
+    host = minimal_host(
+        account=SYSTEM_IDENTITY["account_number"],
+        insights_id=expected_insights_id,
+        reporter="rhsm-conduit",
+        subscription_manager_id=OWNER_ID,
+    )
+
+    message = wrap_message(host.data(), "add_host")
+
+    with pytest.raises(ValidationException):
+        handle_message(json.dumps(message), mock_event_producer)
+
+
```

Comment:
There's a lot of stuff in here that's unnecessary due to what you're testing.

```suggestion
def test_rhsm_reporter_and_no_platform_metadata(mocker, flask_app):
    host = minimal_host(
        account=SYSTEM_IDENTITY["account_number"],
        insights_id=generate_uuid(),
        reporter="rhsm-conduit",
        subscription_manager_id=OWNER_ID,
    )

    message = wrap_message(host.data(), "add_host")

    with pytest.raises(ValidationException):
        handle_message(json.dumps(message), mocker.Mock())


```

Reply:
Fixed using your code.

Reply:
If you did, you didn't push the change @thearifismail